pkg/models/component.go

```diff
@@ -0,0 +1,29 @@
+package models
```

Comment:
I like this, but shouldn't this be in `component` package?
Everything here is related to the component.

Good go practice is that definitions like this are in `types.go` file in the package that they are related to.

Reply:
Yeah but I just did this to avoid cyclic dependencies 
OcClient also has references to this struct which makes occlient depend on package containing this.. and component pkg references occlient

Reply:
than this indicates flaw in our design :-|

How about moving this to `occlient`?

Reply:
OK, I will answer that myself, that is won't be nice :-)

Reply:
Would that be ok ?

Reply:
not ideal :-( ideally this should be in `component` and functions in `occlient` should use a different structure. We need to decide for what those arguments are for. 
Is it for functions for components or for functions for `occlient`?

Reply:
Ok but I felt the same struct can be used instead of creating one more

Reply:
Ideally, we should have nice abstraction levels, where the component is one level and occlient is second. component can use lover level abstractions like occlient. And lower-level abstractions shouldn't know anything about higher levels. We shouldn't import anything from component in occlient. We already broke this multiple times, and this is what is causing all the cyclic imports :-(

Reply:
Having a separate package with just "models" can be tricky.
To make it work you have to limit imports in that package, ideally, don't have any. You are already using `utils` there, that means that now you can't use anything from `models` in utils.

Having a separate package like this also prevents you from creating receiver methods for those types, because you would have to put those functions to the same package, this would resolve into adding more imports .....

Reply:
Hmm so that means https://github.com/redhat-developer/odo/pull/991#discussion_r235377770 makes sense

Reply:
@kadel Done I put it finally in occlient

Its actually almost the same set of params that are used from the cmd layer all the way upto occlient with off-course a few of them not used in some of the occlient functions related to create
but overall its almost the same.. So, based on this discussion I felt it makes sense to move it to occlient..